src/connect.rs

```diff
@@ -236,12 +236,12 @@ impl Connector {
                 }
             }
             #[cfg(feature = "__rustls")]
-            Inner::RustlsTls { tls_proxy, .. } => {
+            Inner::RustlsTls { tls, .. } => {
                 if dst.scheme() == Some(&Scheme::HTTPS) {
                     use std::convert::TryFrom;
                     use tokio_rustls::TlsConnector as RustlsConnector;
 
-                    let tls = tls_proxy.clone();
+                    let tls = tls.clone();
```

Comment:
Shouldn't this still use the proxy TLS? Should the fix instead be that we set ALPN on the proxy TLS?

Reply:
we can see when `connect_via_proxy` and `ProxyScheme::Socks5`, we call `self.connect_socks(dst, proxy_scheme).await`, `dst` is our target. 

```rust
            #[cfg(feature = "__rustls")]
            Inner::RustlsTls { tls, .. } => {
                if dst.scheme() == Some(&Scheme::HTTPS) {
                    use std::convert::TryFrom;
                    use tokio_rustls::TlsConnector as RustlsConnector;

                    let tls = tls.clone();
                    let host = dst.host().ok_or("no host in url")?.to_string();
                    // here I also think it's not suitable to detect proxy port from dst
                    // since socks5 proxy is different from http proxy and it seems
                    // the original contributor took http proxy as reference when implementing
                    // socks proxy
                    let conn = socks::connect(proxy, dst, dns).await?; // create socks tunnel
                    let conn = TokioIo::new(conn);
                    let conn = TokioIo::new(conn);
                    let server_name =
                        rustls_pki_types::ServerName::try_from(host.as_str().to_owned())
                            .map_err(|_| "Invalid Server Name")?;
                    let io = RustlsConnector::from(tls) // here we create TLS connector for HTTPS through socks tunnel, so `tls` should not be proxy TLS.
                        .connect(server_name, conn)
                        .await?;
                    let io = TokioIo::new(io);
                    return Ok(Conn {
                        inner: self.verbose.wrap(RustlsTlsConn { inner: io }),
                        is_proxy: false,
                        tls_info: false,
                    });
                }
            }
```